dts/bindings/interrupt-controller/nordic,split-channels.yaml

```
@@ -0,0 +1,31 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+description: Nordic Split Channels
```

this deserves a better description

---

@adamkondraciuk could you help out here?

---

@gmarull , something like this would be fine?
```suggestion
description: Nordic Split Channels

Some of Nordic's peripherals support split ownership feature that allows to
be used by independent owners. As an example the configuration of the
Global Real Time Counter (GRTC) is shown below:

  owned-channels = <0 1 2 3 4 5 6 7 8 9 10 11>;
  child-owned-channels = <7 8 9 10 11>; 
  
Which means that channels 0-11 will be assigned to the particular CPU.
Other CPUs cannot use those and another set must be defined for them.
In addition, `child-owned-channels` property allows to use channels
7-11 only by child subprocessor. If the CPU you're configuring has no
subprocessor(s) assigned, the `child-owned-channels` property is redundant.
```

---

> the `child-owned-channels` property is redundant

I don't get it. Redundant with what?

---

s/redundant/ignored considered the context

---

Well, it's not really ignored, because `child-owned-channels` are never used by the main CPU. These channels would become unusable if there is no subprocessor assigned.

---

True, so how about something like that:
```suggestion
description: Nordic Split Channels
Some of Nordic's peripherals support split ownership feature that allows to
be used by independent owners. As an example the configuration of the
Global Real Time Counter (GRTC) is shown below:
  owned-channels = <0 1 2 3 4 5 6 7 8 9 10 11>;
  child-owned-channels = <7 8 9 10 11>; 
  
Which means that channels 0-11 will be assigned to the particular CPU.
Other CPUs cannot use those and another set must be defined for them.
In addition, `child-owned-channels` property allows to use channels
7-11 only by child subprocessor. If the CPU you're configuring has no
subprocessor(s) assigned, the `child-owned-channels` property
should not be defined.
```

---

@pdunaj @hubertmis I changed the description to the last one suggested by @adamkondraciuk . If you have any notes on that, please, let me know. Otherwise, resolve this conversation.